### PR TITLE
Make database configuration relative to the workspace root

### LINKED ISSUE #241: Connection use relative path of workspace
**Is your feature request related to a problem? Please describe.**
I am using a sqlite connection, the path to the database file is the full, resolved path.

**Describe the solution you'd like**
Instead of a full path, use a relative path to the workspace. For example, it saved this to `.vscode/settings.json`:

```json
{
  "sqltools.connections": [
    {
      "database": "/Users/my-user/my-project/nested/dir/mydatabase.db",
      "dialect": "SQLite",
      "name": "my project database"
    }
  ]
}
```

If my workspace is opened up at the `/Users/my-user/my-project`, then I'd rather have this:

```json
{
  "sqltools.connections": [
    {
      "database": "./nested/dir/mydatabase.db",
      "dialect": "SQLite",
      "name": "my project database"
    }
  ]
}
```

The reason being, if I check in this `.vscode/settings.json`, my full path is likely not going to match anyone else's full path. The good news is, if I change the path manually, it will actually work just fine, I can connect to the sqlite database file and see the tables. So this is just about what is saved to `.vscode/settings.json`.

Maybe something like this:

```javascript
const path = require('path');
const databasePath = '/Users/my-user/my-project/nested/dir/mydatabase.db';
const database = path.relative(vscode.workspace.rootPath, databasePath);
// database should be "./nested/dir/mydatabase.db"
```

### PR DESCRIPTION
**Pull request by mitchellsimoens**
Fixes #241 to make the database connection config relative to the workspace.

----

Thank you for your contribution! 
Before submitting this PR, please make sure:

- [x] Your code builds clean without any errors or warnings
- [x] You have made the needed changes to the docs
- [x] You have wrote a description of what is the purpose of this pull request above

### CODE REVIEW

**Comment by mitchellsimoens** [comment]
PR #255

File: packages/plugins/connection-manager/screens/settings.ts [hunk 1]
```diff
@@ -28,6 +28,11 @@ export default class SettingsWebview extends WebviewProvider {
 
   private updateConnection = async ({ connInfo, isGlobal, editId }) => {
     console.log({ connInfo, isGlobal, editId });
+
+    if (connInfo.database) {
```
Comment (mtxr) [inline packages/plugins/connection-manager/screens/settings.ts]: We need to check if connInfo.dialect is SQLite.

Adding this change, we are good to go! Thanks @mitchellsimoens !

**Comment by mtxr** [comment]
Thanks @mitchellsimoens! That's awesome! 

I've made just a change to use relative path if it's inside of a workspace.

If the file is somewhere out of a workspace, we use the realpath.

**Comment by mitchellsimoens** [comment]
@mtxr nice final solution! this was my first attempt at vscode extension stuff so thanks for helping me along too! Also, thanks for the work on this extension, I think it's gonna be part of my workflows on current and future projects.

**Pull request merged**